protocol-tests/src/main/java/fixtures/dpgcustomization/implementation/DpgClientImpl.java

```diff
@@ -709,6 +711,51 @@ public DpgClientImpl(HttpPipeline httpPipeline, SerializerAdapter serializerAdap
                                         null));
     }
 
+    /**
+     * Sends the {@code httpRequest}.
+     *
+     * @param httpRequest The HTTP request to send.
+     * @return the response body on successful completion of {@link Mono}.
+     */
+    @ServiceMethod(returns = ReturnType.SINGLE)
+    public Mono<Response<BinaryData>> sendRequestAsync(HttpRequest httpRequest) {
+        return this.getHttpPipeline()
+                .send(httpRequest)
+                .flatMap(
+                        response ->
+                                BinaryData.fromFlux(response.getBody())
+                                        .map(
+                                                body ->
+                                                        new SimpleResponse<>(
+                                                                response.getRequest(),
+                                                                response.getStatusCode(),
+                                                                response.getHeaders(),
+                                                                body)));
+    }
+
+    /**
+     * Sends the {@code httpRequest}.
+     *
+     * @param httpRequest The HTTP request to send.
+     * @param context The context to associate with this operation.
+     * @return the response body along with {@link Response}.
+     */
+    public Response<BinaryData> sendRequest(HttpRequest httpRequest, Context context) {
+        return this.getHttpPipeline()
+                .send(httpRequest, context)
+                .flatMap(
+                        response ->
+                                BinaryData.fromFlux(response.getBody())
+                                        .map(
+                                                body ->
+                                                        new SimpleResponse<>(
+                                                                response.getRequest(),
+                                                                response.getStatusCode(),
+                                                                response.getHeaders(),
+                                                                body)))
+                .block();
```

Comment:
Can we just call the async method and block on it?

Reply:
PR to core https://github.com/Azure/azure-sdk-for-java/pull/27197